cou/steps/analyze.py

```diff
@@ -50,29 +55,33 @@ async def create(cls) -> Analysis:
         return Analysis(apps=apps)
 
     @classmethod
-    async def _populate(cls) -> List[Application]:
+    async def _populate(cls) -> list[OpenStackApplication]:
         """Analyze the applications in the model.
 
         :return: Application objects with their respective information.
-        :rtype: List[Application]
+        :rtype: List[OpenStackApplication]
         """
         juju_status = await async_get_status()
         model_name = juju_status.model.name
         apps = {
-            Application(
+            AppFactory.create(
                 name=app,
                 status=app_status,
                 config=await async_get_application_config(app),
                 model_name=model_name,
+                charm=await extract_charm_name(app),
             )
             for app, app_status in juju_status.applications.items()
         }
-        upgradeable_apps = {app for app in apps if app.charm in UPGRADE_ORDER}
+        # remove non-supported charms that return None on AppFactory.create
+        apps.remove(None)
```

Comment:
```suggestion
        apps.discard(None)
```
You may not have non-OpenStack applications in your model (hence no None members of your set)

Reply:
and sorry for the double double negation :laughing: